scipy/sparse/linalg/interface.py

```diff
@@ -367,6 +367,12 @@ def dot(self, x):
                 raise ValueError('expected 1-d or 2-d array or matrix, got %r'
                                  % x)
 
+    def __matmul__(self, other):
+        if np.isscalar(other):
+            raise ValueError("Scalar operands are not allowed, "
+                             "use '*' instead")
+        return self.__mul__(other)
+
     def __rmul__(self, x):
```

Comment:
Sure – I initially left it out because there's no right-side equivalent of ``dot()``, but I can add it now.